### ISSUE TITLE
try not to call "Require_login" inside webservice functions

### ISSUE THREAD

**Issue by danmarsden** [cursiveinc/moodle-tinymce_cursive#82]
https://github.com/cursiveinc/moodle-tinymce_cursive/blob/main/externallib.php#L72

The docs state:
Note: validate_context() is required in all external functions before operating on any data belonging to a context. This function does sanity and security checks on the context that was passed to the external function - and sets up the global $PAGE and $OUTPUT for rendering return values. Do NOT use require_login(), or $PAGE->set_context() in an external function.

basically require_login can do various things like redirect which you don't want to happen inside the webservice function..

This is not a blocker for approval.

**Comment by nawazsharif** [comment]
@danmarsden
remove the required login from the webservice
You could check now
